compiler/parser/RdrHsSyn.hs

```diff
@@ -2852,14 +2866,74 @@ mkTemplateDecls header fields decls = do
   choiceDataDecls <- concat <$> traverse mkChoiceDataDecls vtChoices
   let templateName = occNameString . rdrNameOcc <$> vtTemplateName
       templateDataDecl = mkTemplateDataDecl (combineLocs vtTemplateName fields) vtTemplateName vtTypeVars ci
+      (letDecl,sharedBinds) = shareTemplateLetBindings conName vtLetBindings
       choicesWithArchive = mkArchiveChoice : vtChoices
-      templateInstances = mkTemplateInstanceDecl templateName conName vt
+      templateInstances = mkTemplateInstanceDecl sharedBinds templateName conName vt
       choiceInstanceDecls = concatMap (mkChoiceInstanceDecl templateName vtTypeVars) choicesWithArchive
-      choiceDecls = concatMap (mkChoiceDecls (getLoc templateName) conName vtLetBindings) choicesWithArchive
-      keyInstanceDecl = mkKeyInstanceDecl templateName conName vt -- <$> kdKeyType . unLoc <$> vtKeyData
-  return $ toOL $ templateDataDecl : choiceDataDecls
+      choiceDecls = concatMap (mkChoiceDecls (getLoc templateName) conName sharedBinds) choicesWithArchive
+      keyInstanceDecl = mkKeyInstanceDecl sharedBinds templateName conName vt -- <$> kdKeyType . unLoc <$> vtKeyData
```

Comment:
Do you know what the point of the type and data are here? If there is none, I would just remove it.

Reply:
I have no idea.  I removed it.

Reply:
`collectLocalBinds` takes `HSLocalBinds` (which I have) instead of `HsValBindLR` which is one step within